numpyro/util.py

```diff
@@ -392,6 +392,11 @@ def soft_vmap(fn, xs, batch_ndims=1, chunk_size=None):
     ys = lax.map(fn, xs) if num_chunks > 1 else fn(xs)
     map_ndims = int(num_chunks > 1) + int(chunk_size > 1)
     ys = tree_map(
-        lambda y: jnp.reshape(y, (-1,) + jnp.shape(y)[map_ndims:])[:batch_size], ys
```

Comment:
Good catch! When `y.size` is `0`, we can't use `(-1)` to reshape it. I guess we can also do:
```jnp.reshape(y, (np.prod(jnp.shape(y)[:map_ndims]),) + jnp.shape(y)[map_ndims:])[:batch_size]```
here to preserve the batch dimensions of y.